node/src/tasks/bsp_charge_fees.rs

```diff
@@ -104,3 +198,197 @@ where
         Ok(())
     }
 }
+
+impl<FL, FSH> EventHandler<UserWithoutFunds> for BspChargeFeesTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: BspForestStorageHandlerT,
+{
+    async fn handle_event(&mut self, event: UserWithoutFunds) -> anyhow::Result<()> {
```

Comment:
This task is not filtering for a user that this BSP is serving. That means that this will queue up `StopStoringForInsolventUserRequest`s for every single user without funds, regardless of whether it can charge it or not.

Reply:
Yes, how it works right now is that the handler of the actual deletion process checks the forest storage to see if there's any file belonging to that user, but I see that it can be more efficient to check before since we don't need to wait to acquire a write lock on the forest root (since we are only reading) so I modified both runtime event handlers to do exactly that

Reply:
Exactly, that was my point, avoiding having to acquire the write lock.